rest/permissions.py

```diff
@@ -0,0 +1,46 @@
+"""
+Permission classes for REST views
+"""
+
+from __future__ import unicode_literals
+
+from rest_framework.permissions import (
+    BasePermission,
+    SAFE_METHODS,
+)
+from django.http import Http404
+
+from learningresources.models import Repository
+from learningresources.api import (
+    get_repo,
+    PermissionDenied,
+    NotFound,
+)
+
+
+# pylint: disable=protected-access
+class AddRepoPermission(BasePermission):
```

Comment:
It's difficult to name because permission classes need to handle both read and write methods. If we wanted separate permission classes for different methods then we can override `get_permissions()` on the view to check `request.method` there instead, but then we're just moving the work. The [Django REST framework docs](http://www.django-rest-framework.org/api-guide/permissions/#custom-permissions) recommend doing it in the permission class although at the moment I'm not sure how reusable these permission classes will be.

We could override `check_permissions` in each view and just do the checking there instead, if it's unlikely we can make reusable permission classes.